### PR TITLE
Implement per-user package directory feature

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR resolves issue lablup/backend.ai#99.

### CODE REVIEW

File: src/ai/backend/agent/docker/agent.py [hunk 1]
```diff
@@ -546,7 +546,18 @@ def __init__(self, config, *, skip_initial_scan: bool = False) -> None:
...
                 mounts.append(mount)
-
+            if package_directory := kernel_config['package_directory']:
+                folder_id, group_username, folder_host = package_directory
+                kernel_path = Path('/home/work/.local')
+                package_host_path: Path = (self.config['vfolder']['mount'] / folder_host /
+                             self.config['vfolder']['fsprefix'] / folder_id)
+                if group_username:
+                    package_host_path = package_host_path / group_username
+                package_host_path.mkdir(parents=True, exist_ok=True)
```
Comment (achimnol) [inline src/ai/backend/agent/docker/agent.py]: When agents are running as root, wouldn't it be necessary to `chown` the newly created directories under vfolders?
(This is the first case where agents directly manipulate the content of vfolders!)

File: src/ai/backend/agent/docker/agent.py [hunk 2]
```diff
@@ -546,7 +546,18 @@ def __init__(self, config, *, skip_initial_scan: bool = False) -> None:
                 mount = Mount(MountTypes.BIND, host_path, kernel_path, folder_perm)
                 resource_spec.mounts.append(mount)
                 mounts.append(mount)
-
+            if package_directory := kernel_config['package_directory']:
```
Comment (achimnol) [inline src/ai/backend/agent/docker/agent.py]: If needed, update this line according to the fixes for the manager-side PR.

File: src/ai/backend/agent/docker/agent.py (hunk 1 above)
Comment (achimnol) [reply src/ai/backend/agent/docker/agent.py]: I also believe that manipulation of vfolder contents should be done in the manager for the consistency with existing codebase and reduced possibility of bugs ("gathering related things in one place" principle).

**Comment by kyujin-cho** [comment]
Since all changes in this PR has been migrated to manager's PR, I'll close this PR.

**Pull request closed without merging**